First new gateway in a year.

Yobrocharlie opened the pull request:
## About The Pull Request

This PR adds a new gateway with not too much threat a, a little bit of reward more often in the form of gimmicky items or armor.

credit to https://github.com/NovaSector/NovaSector/pull/4880
For the gun and sfx sprite
## Why It's Good For The Game

The gateway system is about as decrepit and old, the last gateway that was added was the museum which was made by somebody who would short after receive a rule 8 ban and plus that away mission is a neat scenery and has a lot of cool locations to look at but it has no loot, and you know what? that's okay and it achieves what it sets out to do beautifully, but if we discount that one for that reason alone (which for the record is a bad reason to discount something) then the last gateway that was added and that stuck was added in 2018? and it was snowdin? I dont actually know, correct me on this. But basically, more content good.

## Changelog

<details>
<summary>Screenshots/Videos</summary>
<img width="862" height="431" alt="image" src="https://github.com/user-attachments/assets/d56add04-4b8c-4cd0-be7f-36d895e43868" />
<img width="654" height="665" alt="image" src="https://github.com/user-attachments/assets/1ccd15c4-5c31-4b00-b5f8-9a662efa3436" />
<img width="651" height="571" alt="image" src="https://github.com/user-attachments/assets/08ab4c1d-d9d0-413c-bc02-a3c51eda8761" />
<img width="1177" height="172" alt="image" src="https://github.com/user-attachments/assets/8df5e28d-5041-4676-8370-b8ab351f5b52" />
<img width="850" height="586" alt="image" src="https://github.com/user-attachments/assets/c03e6cbe-b290-421e-bbed-80a203221de4" />
<img width="1248" height="912" alt="image" src="https://github.com/user-attachments/assets/aa3c415d-52de-46b9-b317-e397b481b1b7" />
<img width="885" height="697" alt="image" src="https://github.com/user-attachments/assets/dd96d364-deae-4c56-9fc1-68b5e738ef4a" />
<img width="1284" height="646" alt="image" src="https://github.com/user-attachments/assets/b433f591-d92f-4bea-921c-b231fce2e7d9" />

also NO you cannot make the sm sword using the slow recharge shrink ray
<img width="621" height="240" alt="image" src="https://github.com/user-attachments/assets/75f7d235-33c2-424e-9260-a3cc8c931ab9" />
</details>

<details>
<summary>Documentation of all notable loot 11/8/2025</summary>
  /obj/item/clothing/shoes/gunboots/disabler
  /obj/item/gun/energy/shrink_ray/one_shot (A shrink way that has a long recharge time and can only shoot once)
  /obj/item/slimepotion/lovepotion x 2
  /obj/item/slimepotion/genderchange x 1
  /obj/item/slimepotion/slime/mutator
  /obj/item/gun/energy/disabler/smoothbore
  /obj/item/gun/energy/alien/zeta
  /obj/item/gun/magic/midas_hand
  /obj/item/autosurgeon/xeno (A autosurgeon that only allows xenomorph organs, here because it looks unique)
  /obj/item/organ/heart/gland (abductor organs, random.)
  /obj/item/organ/eyes/corrupt
  /obj/item/organ/appendix/corrupt
  /obj/item/clothing/suit/utility/radiation/cbrnsuit x 2
  /obj/item/clothing/head/utility/radiation/cbrnhood x 2
  /obj/item/clothing/gloves/combat x 4
  /obj/item/clothing/shoes/combat/swat x 2
  /obj/item/clothing/mask/gas/atmos/glass x 2 
  /obj/item/clothing/under/rank/engineering/atmospheric_technician/nova/utility/advanced
  /obj/item/gun/ballistic/automatic/wt550/p90 x 2
  /obj/item/ammo_box/magazine/p90_mag x 14
  /obj/item/gun/energy/modular_laser_rifle/carbine x 2
  /obj/effect/spawner/random/armory/laser_gun
  /obj/item/grenade/c4 x 3
  /obj/machinery/deployable_turret/hmg x 3 (you think these are good, but they're not, they are really not, they're only good if you hit somebody with them when they're not expecting it, otherwise people will simply just avoid it, or if your being defensive of an area this thing will have you just be a beast.)
  /obj/machinery/mounted_machine_gun x 8 (But with no ammo, so theyre only setpieces, okay well there IS ammo on the map but its anchored, and theres no way to unanchor anchored items, if fallout thirteen players cant do it over the course of 4 years then tg players sure cant.)
  
Also hello there it looks like you actually read this section, if you have dont worry, this PR wont get merged, why? I dont want to salt about that here.
</details>

:cl:
map: added the "heretic" gateway map (it has nothing related to heretics)
/:cl:

Yobrocharlie committed:
take one

Yobrocharlie committed:
Update away_roles.dm

mikederkan commented:
Will you add images of the gateway?

Yobrocharlie committed:
Update tgstation.dme

HalfPunch commented:
/obj/item/gun/energy/shrink_ray/one_shot

Isn't it a bit too good for a gateway? It's free SM sword if gate spawned

Yobrocharlie committed:
okay

Yobrocharlie commented:
> /obj/item/gun/energy/shrink_ray/one_shot
> 
> Isn't it a bit too good for a gateway? It's free SM sword if gate spawned

the crafting doesnt work like that smartie.

HalfPunch commented:
> the crafting doesnt work like that smartie.

1. No need to be cocky over questions
2. Would be more helpful to link code that would answer my question. No need for that now, since I've found it myself, thought that "small" subtype of SM crystal spawns in when station SM recieves "/datum/component/shrink" component, and not from a recipe

Yobrocharlie commented on code/modules/awaymissions/mission_code/heretic/heretic_gateway_tram.dm:
```diff
@@ -0,0 +1,29 @@
+	configured_transport_id = HERETIC_LINE_1
+
+/obj/machinery/computer/tram_controls/heretic_tram_computer
+	icon = 'modular_zubbers/icons/obj/tram/heretic_tram.dmi'
```
a

Yobrocharlie committed:
Update heretic_gateway_tram.dm

Yobrocharlie committed:
Update heretic.dmm

Yobrocharlie committed:
a

Yobrocharlie committed:
no more p90 sprite

MrMelbert commented:
Crew obtainable shrink ray is in fact an evil evil thing (both for validhunting and for antags)

Yobrocharlie committed:
whoops

Yobrocharlie committed:
Update bruceguns.dm

Yobrocharlie committed:
it cant be this simple.

Yobrocharlie committed:
Update heretic_gateway_misc.dm

necromanceranne reviewed (changes requested):
It seems we couldn't really come to a proper conclusion about the addition of a new weapon here, but I'm going to be more firm in saying I just don't like the inclusion of another ballistic weapon without a distinct purpose.

This particular one already on the PR cannot be included as it will definitely cause performance problems (god I hate full auto guns). A 0.05 fire rate is just...a lag machine. Also making it just the M90gl again but with a different ammo type is probably also not going to fly. 

But it seems we didn't really get anywhere talking about alternatives. I'm just strongly voicing my displeasure about the whole thing.

I'll at least leave my thoughts on the matter and say that any weapons you include, it'd be much more interesting if you give it a relevant context to the place you can find it and the story you're trying to tell with its inclusion. New is not automatically better. And if this is heretic themed, I think leaning into that might be more interesting.

Kiromaka commented:
> Crew obtainable shrink ray is in fact an evil evil thing (both for validhunting and for antags)

Finally crew obtainable SM sword, after (however long ago the craft was added.)

mc-oofert commented:
mapdiffbot picture (hope this embeds)

![picture](https://affectedarc07.blob.core.windows.net/mdb2/images/3234987/55073348691/a/_maps_RandomZLevels_heretic/0-added.png)

Yobrocharlie committed:
Fake_items

Yobrocharlie committed:
Update tgstation.dme

Yobrocharlie committed:
Update heretic.dmm

Yobrocharlie committed:
Update fake_items.dm

Yobrocharlie committed:
no more p90

Yobrocharlie committed:
tg players cant be trusted with a ghost role like this, and im not going to make it mechanically enforced.

Yobrocharlie committed:
removing the ghost role spot

Yobrocharlie committed:
fixes the napad

Yobrocharlie committed:
here wwe go

Yobrocharlie committed:
better lighting

HalfPunch commented:
> If fallout thirteen players cant do it over the course of 4 years then tg players sure cant

Bet

Yobrocharlie commented:
> > If fallout thirteen players cant do it over the course of 4 years then tg players sure cant
> 
> Bet

if you can figure out a method that any player can pull off ill give you 1 month nitro

Yobrocharlie committed:
Your not worth the effort

I cant figure out how to make the sprite work, and even if i did people would only complain about this anyhow thinking about how it would be the real thing or how it could be abused (like any other megafauna) and all of that simply is not worth the idea of this being a setpiece.

Yobrocharlie commented on sound/items/weapons/gun/rifle/smg_heavy.ogg:
A

Yobrocharlie replied:
I am currently un-able to access my repo for the next while- if somebody could give proper attribution to trelus added in PR https://github.com/NovaSector/NovaSector/pull/4880 that would be greatly appriciated

Yobrocharlie committed:
Create attributions.txt

Yobrocharlie committed:
fix and message fix

Yobrocharlie committed:
Update heretic.dmm

Yobrocharlie committed:
okay

Yobrocharlie committed:
Update heretic.dmm

Yobrocharlie committed:
Decal update

Yobrocharlie committed:
More medbay decal

Yobrocharlie committed:
Update heretic.dmm

Yobrocharlie committed:
Update heretic.dmm

Yobrocharlie committed:
whoops

Yobrocharlie committed:
Update heretic.dmm

Yobrocharlie committed:
Update heretic.dmm

Maurukas reviewed (approved):
I had fun with it. We need more of these, and there are a lot of bits I like here - the randomly spawning keycards, the tram. There's some areas that are a little low on detail, but others I quite like. 

So let's do it. 

Any loot is subject to admin complaints and the like.
